modules/prebidServerBidAdapter/index.js

```diff
@@ -294,6 +294,41 @@ function transformHeightWidth(adUnit) {
   return sizesObj;
 }
 
+function addFirstPartyDataToRequest(request) {
+  const bidderConfig = config.getBidderConfig();
+  let context = {};
+  let user = {};
+  const allowedBidders = [];
+  Object.keys(bidderConfig).forEach(bidder => {
+    const currBidderConfig = bidderConfig[bidder];
+    if (currBidderConfig.fpd) {
+      allowedBidders.push(bidder);
+    } else {
+      return;
+    }
+
+    if (currBidderConfig.fpd.context) {
+      context = Object.assign({}, context, currBidderConfig.fpd.context);
+    }
+
+    if (currBidderConfig.fpd.user) {
+      user = Object.assign({}, user, currBidderConfig.fpd.user);
+    }
+  });
```

Comment:
Looping over all the bidder data and assigning it to a singular data structure means overwriting all the previously set bidder data. Is that the intended implementation?

e.g. if rubicon has `fpd.user.id = 1` and appnexus has `fpd.user.id = 2` then one will overwrite the other. Is there cases where first party data will be different among bidders?

Reply:
Ugh. We didn't properly specify how to handle bidder-specific data for Prebid Server. The only scenarios we defined support for is where FPD is either entirely global or entirely bidder-specific.

The [intent-to-implement](https://github.com/prebid/Prebid.js/issues/3687) doesn't cover the scenario where there's global FPD available to all bidders but bidder-specific FPD as well.

I'll post a proposal to that thread and see if we can salvage this.